cmd/restic/cmd_rewrite.go

```diff
@@ -217,6 +217,31 @@ func filterAndReplaceSnapshot(ctx context.Context, repo restic.Repository, sn *r
 	sn.Original = sn.ID()
 	sn.Tree = &filteredTree
 
+	sn.Summary.TotalFilesProcessed = 0
+	sn.Summary.TotalBytesProcessed = 0
+	processNode := func(_ restic.ID, nodepath string, node *restic.Node, err error) error {
```

Comment:
Recalculating the snapshot size is rather expensive and only makes sense if `len(rejectByNameFuncs) > 0`.

Iterating through the tree a second time is rather inefficient, instead this recalculation must happen in the `TreeRewriter` in internal/walker/rewriter.go (but only if `DisableNodeCache` is true) where it can be done nearly for free.

Reply:
By updating snapshot summaries of all snapshots that have a certain path, would we change snapshots we initially didn't mean to touch. I think the recalculating of the size should only be relevant to the specified snapshot. No reference of snapshot gets passed into the `TreeRewriter`.

Reply:
Would referencing snapshots in `internal/walker/rewriter.go` cause a circular referencing? Maybe we should update in the `RewriteNode` callback like in the latest commit.

Reply:
I suppose the execution takes place in `internal/walker/rewriter.go`, haha.